src/iqstat.cpp

```diff
@@ -357,3 +234,137 @@ void usage() {
+    genQueryInp_t genQueryInp{};
+    genQueryOut_t *genQueryOut{};
+
+    constexpr output_format_parameters output_format;
+
+    auto num_cols_selected = (brief ? output_format.brief_format_len
+                                    : output_format.long_format_len);
+
+    for (int i = 0; i < num_cols_selected; i++) {
+        addInxIval(&genQueryInp.selectInp, output_format.columns[i], 0);
+    }
+
+
+    if (!allFlag) {
+        char v1[BIG_STR];
+        snprintf( v1, BIG_STR, "='%s'", userName );
+        addInxVal(&genQueryInp.sqlCondInp, COL_RULE_EXEC_USER_NAME, v1);
+    }
+
+    std::string diagnostic;
+    if (ruleName != NULL && *ruleName != '\0') {
+        char v2[BIG_STR];
+        diagnostic = (boost::format( " (and matching key '%s')" ) % ruleName).str();
+        snprintf( v2, BIG_STR, "='%s'", ruleName );
+        addInxVal(&genQueryInp.sqlCondInp, COL_RULE_EXEC_ID, v2);
+    }
+
+    genQueryInp.maxRows = 50;
+
+    // idempotent version of clearGenQueryOut.
+    auto clear_gq_out = [&o = genQueryOut]{ clearGenQueryOut(o);
+                                            o->attriCnt = 0;
+                                          };
+
+    // TODO's : should we set attriCnt = 0 as the last thing in the clearGenQueryOut() routine?
+    // And...investigate 1. if this ^^^ is the best way or via init'ing via memset(ptr,0,sizeof(...)) 
+    //                   2. reimplementation of iqstat using the C++ query iterator
+    //                   3. possibility of memory leaks on rcGenQuery/rsGenQuery side
+
+    // RAII for both GenQuery variables.
+    auto cleanup_gq_in_out = [&, &i = genQueryInp, &o = genQueryOut]{ clearGenQueryInp(&i);
+                                                                      clear_gq_out();
+                                                                      free(o); o = nullptr; };
+    // Deploy the RAII.
+    const auto cleanup_gq = irods::at_scope_exit{[&] { cleanup_gq_in_out(); }};
+
+    int status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+
+    if ( status == CAT_NO_ROWS_FOUND ) {
+        // Need to determine which table "no rows" refers to: USER or RULE.
+        namespace ia = irods::experimental::administration;
+        if (ia::client::exists(*Conn, ia::user{userName})) {
+            if ( allFlag ) {
+                printf( "No delayed rules pending%s\n", diagnostic.c_str() );
+            }
+            else {
+                printf( "No delayed rules pending for user %s%s\n", userName, diagnostic.c_str());
+            }
+            return 0;
+        }
+        else {
+            printf( "User %s does not exist.\n", userName );
+            return 0;
+        }
+    }
+    else if (status < 0) {
+        char* errno_message{};
+        const char* main_message = rodsErrorName(status, &errno_message);
+        std::cerr << "Error in rcGenQuery: " << main_message;
+        if (errno_message && *errno_message) {
+            std::cerr << " - "s + errno_message;
+        }
```

Comment:
Will the error message end up looking like this if `errno_message` has contents?
`"Error in rcGenQuery:  - <message goes here>"`
Looks kind of funny to me. Maybe the " - " or the colon following the message on 311 can be omitted?

Reply:
I'm sure Error in rcGenQuery will always have an all caps error code immediately after, which is contained in the main_message variable.  

Do we have a convention for how the errno sub-message is printed?  `ERRCODE ( errno-msg )` or maybe `ERRCODE: errno-msg` ?

Reply:
Sorry, I meant to say "if `errno_message` has *no* contents". In any case, if that happens, then something has gone really sideways, based on what you're saying. This is fine as-is, then.

Reply:
I think this is working now more in the manner of other icommands' server status reporting. Tested manually using gdb.